Author: Tiago de Paula <tiagodepalves@gmail.com>
Date: Sun, 15 Jun 2025 01:13:10 -0300
Last-Update: 2026-08-31
Subject: Revert "Close file descriptors after startup. Fixes #333186."

Allow showing logs on journalctl.

This reverts commit 2bec7e4653f672a51ff4780d4ff68d73af026e77.
---
 esd.c |   12 ------------
 1 file changed, 12 deletions(-)

Index: b/esd.c
===================================================================
--- a/esd.c
+++ b/esd.c
@@ -632,9 +632,6 @@ int main ( int argc, char *argv[] )
     char *hostname=NULL;
     char *endptr;
 
-    int fd0;
-    int max_fds;
-
     /* from esd_config.c */
     extern char esd_spawn_options[];
     extern char esd_default_options[];
@@ -836,15 +833,6 @@ int main ( int argc, char *argv[] )
     /* cd to / */
     chdir ("/");
 
-    /* close all open file descriptors */
-    max_fds = getdtablesize();
-    for(i=0;i<max_fds;i++){
-      close(i);
-    }
-    fd0 = open("/dev/null", O_RDWR);
-    dup(0);
-    dup(0);
-
     /* open the listening socket */
   listen_socket = open_listen_socket(hostname, esd_port );
   if ( listen_socket < 0 ) {
